lib/spack/spack/cmd/buildcache.py

```diff
@@ -255,10 +255,11 @@ def setup_parser(subparser: argparse.ArgumentParser):
 
     # Sync buildcache entries from one mirror to another
     sync = subparsers.add_parser("sync", help=sync_fn.__doc__)
-    sync.add_argument(
-        "--manifest-glob", help="a quoted glob pattern identifying copy manifest files"
+    sync_source = sync.add_mutually_exclusive_group(required=True)
```

Comment:
I'm just curious if making `--manifest-glob` and `src-mirror` mutually exclusive is fundamentally related to allowing an arbitrary destination, or just something added incidentally.

Reply:
previously this was a "bug" where you could specify both, but the `src_mirror` argument was implicitly ignored. I just made it explicit here.